clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,318 @@
+
+export default class ApiService {
+  private context: string;
+  private hostUrl: string;
+  private userId: string;
+  private token: string;
+  private apiVersion: string;
+  private clientSessionId: string;
+  private httpClient: UpGradeClientInterfaces.IHttpClientWrapper;
+  private api: IEndpoints;
+
+  constructor(config: UpGradeClientInterfaces.IConfig, private dataService: DataService) {
+    this.context = config.context;
+    this.hostUrl = config.hostURL;
+    this.token = config.token;
+    this.clientSessionId = config.clientSessionId;
+    this.userId = config.userId;
+    this.apiVersion = config.apiVersion;
+    this.api = {
+      init: `${this.hostUrl}/api/${this.apiVersion}/init`,
+      getAllExperimentConditions: `${this.hostUrl}/api/${this.apiVersion}/assign`,
+      markDecisionPoint: `${this.hostUrl}/api/${this.apiVersion}/mark`,
+      setGroupMemberShip: `${this.hostUrl}/api/${this.apiVersion}/groupmembership`,
+      setWorkingGroup: `${this.hostUrl}/api/${this.apiVersion}/workinggroup`,
+      failedExperimentPoint: `${this.hostUrl}/api/${this.apiVersion}/failed`,
+      getAllFeatureFlag: `${this.hostUrl}/api/${this.apiVersion}/featureflag`,
+      log: `${this.hostUrl}/api/${this.apiVersion}/log`,
+      logCaliper: `${this.hostUrl}/api/${this.apiVersion}/log/caliper`,
+      altUserIds: `${this.hostUrl}/api/${this.apiVersion}/useraliases`,
+      addMetrics: `${this.hostUrl}/api/${this.apiVersion}/metric`,
+    };
+    this.httpClient = this.setHttpClient(config.httpClient);
+  }
+
+  private setHttpClient(httpClient: UpGradeClientInterfaces.IHttpClientWrapper) {
+    if (USE_CUSTOM_HTTP_CLIENT && !httpClient) {
+      throw new Error(
+        'Please provide valid httpClient, or use the default (non-"lite") version of the library to our default httpClient.'
+      );
+    }
+
+    if (!USE_CUSTOM_HTTP_CLIENT && httpClient) {
+      throw new Error('Please import "lite" version of the to use custom httpClient.');
+    }
+
+    if (USE_CUSTOM_HTTP_CLIENT && httpClient) {
+      return httpClient;
+    }
+
+    if (!USE_CUSTOM_HTTP_CLIENT && !httpClient) {
+      return new DefaultHttpClient();
+    }
+  }
+
+  private validateClient() {
+    if (!USE_CUSTOM_HTTP_CLIENT && !this.hostUrl) {
+      throw new Error('Please set application host URL first.');
+    }
+    if (!this.userId) {
+      throw new Error('Please provide valid user id.');
+    }
+    if (!this.context) {
+      throw new Error('Please provide valid context.');
+    }
+    if (!this.httpClient) {
+      throw new Error('HttpClient is not defined.');
+    }
+  }
+
+  private createOptions(url: string): UpGradeClientInterfaces.IHttpClientWrapperRequestConfig {
+    const options: UpGradeClientInterfaces.IHttpClientWrapperRequestConfig = {
+      headers: {
+        'Content-Type': 'application/json',
+        'Session-Id': this.clientSessionId || uuid.v4(), // set this here? Do we need to require a clientSessionId?
```

Comment:
Creating uuid here will create new id for each request.

Reply:
updated